creator-node/src/services/sync/processSync.js

```diff
@@ -6,6 +6,48 @@ const { saveFileForMultihashToFS } = require('../../fileManager')
 const { getOwnEndpoint, getCreatorNodeEndpoints } = require('../../middlewares')
 const SyncHistoryAggregator = require('../../snapbackSM/syncHistoryAggregator')
 const DBManager = require('../../dbManager')
+const CIDFailureCountManager = require('./CIDFailureCountManager')
+
+/**
+ * Wrapper around `saveFileForMultihash()` with CID failure count tracking
+ * Throws error if CID failure count is below threshold, else returns boolean indicating saved or skipped
+ * @returns {Boolean} skipped
+ */
+async function saveFileForMultihashToFSWrapper ({
+  serviceRegistry, logger, multihash, expectedStoragePath, gatewaysToTry, fileNameForImage = null, SyncMaxCIDFailureCountBeforeSkip
+}) {
+  let skipped = false
+
+  try {
+    await saveFileForMultihashToFS(
+      serviceRegistry, logger, multihash, expectedStoragePath, gatewaysToTry, fileNameForImage
+    )
+
+    // Reset CID failure count on success
+    CIDFailureCountManager.resetCIDFailureCount(multihash)
+
+    // If saveFileForMultihashToFS throws, it is because of failed content retrieval or verification
+  } catch (e) {
+    // Increment CID failure count
+    CIDFailureCountManager.incrementCIDFailureCount(multihash)
+
+    // If CID failure count is below threshold, throw an error (base case)
+    if (CIDFailureCountManager.getCIDFailureCount(multihash) < SyncMaxCIDFailureCountBeforeSkip) {
+      throw new Error(e)
+
+      // If threshold met, sync should skip CID, indicate as skipped without erroring
+    } else {
+      skipped = true
```

Comment:
you could just return `true` here instead of keeping a var

Reply:
yeah i could, idk i find the named return var makes this more readable
obv its nbd either way, but i'm curious about your thinking

Reply:
not worth too much thought. technically cheaper 🤷

Reply:
yeah ofc its super nitty - i generally opt for more verbose code where possible esp for negligible perf hits like this 🤷‍♂️